irs/ports/outbound/event_pub.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Interfaces for event publishing adapters and the exception they may throw."""
+
+from datetime import datetime
+from typing import Protocol
+
+
+class EventPublisherPort(Protocol):
+    """An interface for an adapter that publishes events happening to this service."""
```

Comment:
We should consistently use the inheritance based ABC for model relationships between 3hex components. (Please let me know, if I am stilll using protocols somewhere in the other repos that should be changed.)

```suggestion
from abc import ABC, abstractmethod


class EventPublisherPort(ABC):
    """An interface for an adapter that publishes events happening to this service."""
```